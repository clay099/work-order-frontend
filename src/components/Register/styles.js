import { makeStyles } from "@material-ui/core/styles";
import { paper, avatar, userType } from "../../styles/styleObjects";

const useStyles = makeStyles((theme) => {
	let paperObj = paper(theme);
	let avatarObj = avatar(theme);
	let userTypeObj = userType(theme);
	return { ...paperObj, ...avatarObj, ...userTypeObj };
});

export default useStyles;
