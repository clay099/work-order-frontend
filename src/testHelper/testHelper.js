// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
