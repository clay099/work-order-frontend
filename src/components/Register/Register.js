import React from "react";
import RegisterForm from "../RegisterForm/RegisterForm";
import {
	CssBaseline,
	Container,
	Typography,
	Avatar,
	Button,
	Grid,
} from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import LockOutlinedIcon from "@material-ui/icons/LockOutlined";
import useToggle from "../../hooks/useToggle";

const useStyles = makeStyles((theme) => ({
	paper: {
		marginTop: theme.spacing(2),
		display: "flex",
		flexDirection: "column",
		alignItems: "center",
	},
	avatar: {
		margin: theme.spacing(1),
		backgroundColor: theme.palette.secondary.main,
	},
	userType: {
		marginTop: theme.spacing(1),
	},
}));

const Register = () => {
	const classes = useStyles();

	const { setTrue: showUser, setFalse: hideUser, toggle } = useToggle(true);

	let userVariant;
	let tradesmenVariant;

	if (toggle) {
		userVariant = "contained";
		tradesmenVariant = "outlined";
	} else {
		userVariant = "outlined";
		tradesmenVariant = "contained";
	}

	return (
		<Container component="main" maxWidth="xs">
			<div className={classes.paper}>
				<CssBaseline />
				<Avatar className={classes.avatar}>
					<LockOutlinedIcon />
				</Avatar>
				<Typography component="h1" variant="h5">
					Sign up
				</Typography>
				<Grid spacing={2} container className={classes.userType}>
					<Grid item xs={12} sm={6}>
						<Button
							fullWidth
							variant={userVariant}
							color="primary"
							onClick={showUser}
						>
							New User
						</Button>
					</Grid>
					<Grid item xs={12} sm={6}>
						<Button
							fullWidth
							variant={tradesmenVariant}
							color="primary"
							onClick={hideUser}
						>
							New Tradesmen
						</Button>
					</Grid>
				</Grid>
				{toggle ? (
					<RegisterForm userType="user" />
				) : (
					<RegisterForm userType="tradesmen" />
				)}
			</div>
		</Container>
	);
};

export default Register;
