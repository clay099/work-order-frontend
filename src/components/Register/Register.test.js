import React from "react";
import Register from "./Register";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

describe("Register Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<Register />);
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<Register />);
		expect(asFragment).toMatchSnapshot();
	});

	test("value to be in the document", () => {
		const { getByText } = render(<Register />);
		const value = getByText("Sign up");
		expect(value).toBeInTheDocument();
	});
});
