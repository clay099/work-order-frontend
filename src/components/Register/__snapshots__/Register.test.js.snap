// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Component matches snapshot 1`] = `[Function]`;
